src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/grid.js

```diff
@@ -285,6 +287,12 @@ define([
           actionConfiguration = row.model.get('action_configuration');
         if (!actionConfiguration || actionConfiguration[action.name] !== false) {
           action.run();
+
+          if (action.datagrid && action.datagrid.entityHint === 'product') {
+            analytics.track('product-grid:product:selected', {
+              name: action.model.attributes.identifier,
+            });
```

Comment:
The grid module is not only used with product grid. We also test the name of the datagrid to be sure. Or, we can define a more generic event and pass the name of the datagrid.

```suggestion
          if (this.name === 'product-grid' && action.datagrid && action.datagrid.entityHint === 'product') {
            analytics.track('product-grid:product:selected', {
              identifier: action.model.attributes.identifier,
            });
```